### PR TITLE
Trace SSL handshakes in netty 4.0

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Continuation of https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4604

### CODE REVIEW

File: instrumentation/netty/netty-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_0/BootstrapInstrumentation.java [hunk 1]
```diff
@@ -32,16 +31,16 @@
   @Override
   public void transform(TypeTransformer transformer) {
     transformer.applyAdviceToMethod(
-        named("doConnect")
-            .and(takesArgument(0, SocketAddress.class))
-            .and(returns(named("io.netty.channel.ChannelFuture"))),
+        named("doConnect0")
```
Comment (iNikem) [inline instrumentation/netty/netty-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_0/BootstrapInstrumentation.java:34]: Just out of curiosity - why this change?

**Review by iNikem — COMMENTED** [review]
Have we already discussed having this as event as opposed to a span? Is there any separation between start/end for that span?

File: instrumentation/netty/netty-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_0/BootstrapInstrumentation.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/netty/netty-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_0/BootstrapInstrumentation.java:34]: The `doConnect` actually traces more than just connection - it starts with tracing channel registration (which happens before) and ends after handler `channelActive()` has returned (which is way after SSL handshake is finished). The `doConnect0` method seems to be doing just the connection, which is more accurate.

**Comment by mateuszrzeszutek** [comment]
> Have we already discussed having this as event as opposed to a span? Is there any separation between start/end for that span?

We've briefly touched upon this idea in the last PR, and we have https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4617 where we can discuss it further. To sum up the previously mentioned points: SSL handshake takes time and has non-zero duration, and can fail - we can record the exception and set the span status.

**Review by iNikem — APPROVED** [review]
